src/components/id3_tag.rs

```diff
@@ -102,38 +106,23 @@ impl AudioTagEdit for Id3v2Tag {
         self.inner.remove_artist();
     }
 
-    fn year(&self) -> Option<i32> {
-        if self.inner.version() == Version::Id3v23 {
-            if let ret @ Some(_) = self.inner.year() {
-                return ret;
-            }
-        }
+    fn date(&self) -> Option<Timestamp> {
+        self.inner.date_recorded()
+    }
+    fn set_date(&mut self, timestamp: Timestamp) {
+        self.inner.set_date_recorded(timestamp)
+    }
+    fn remove_date(&mut self) {
+        self.inner.remove_date_recorded()
+    }
 
-        self.inner.date_recorded().map(|timestamp| timestamp.year)
+    fn year(&self) -> Option<i32> {
+        self.inner.year()
     }
     fn set_year(&mut self, year: i32) {
-        if self.inner.version() == Version::Id3v23 {
-            self.inner.set_year(year);
-            return;
-        }
-
-        if let Some(mut timestamp) = self.inner.date_recorded() {
-            timestamp.year = year;
-            self.inner.set_date_recorded(timestamp);
-            return;
-        }
-
-        self.inner.set_date_recorded(Timestamp {
-            year,
-            month: None,
-            day: None,
-            hour: None,
-            minute: None,
-            second: None,
-        });
+        self.inner.set_year(year);
     }
     fn remove_year(&mut self) {
-        self.inner.remove_date_recorded();
```

Comment:
This should be good once this is added back. I missed this in the review. :)

Reply:
Aded back